include/xtensor/xstrided_view.hpp

```diff
@@ -807,6 +807,56 @@ namespace xt
         );
     }
 
+    namespace detail
+    {
+        template <typename S, typename Enable = void>
+        struct rebind_shape;
+
+        template <class S>
+        struct rebind_shape<S, std::enable_if_t<!std::is_signed<get_value_type_t<typename std::decay_t<S>>>::value>>
+        {
+            using type = S;
+        };
+
+        template <class S>
+        struct rebind_shape<S, std::enable_if_t<std::is_signed<get_value_type_t<typename std::decay_t<S>>>::value>>
+        {
+            using type = rebind_container_t<size_t, S>;
+        };
```

Comment:
Not sure that you need this, I think you can use `rebind_container_t<size_t, S>` anywhere you need, it will be a "noop" for shape holding `size_t`. Unless you want to avoid rebinding shapes whose unsigned type is not `size_t`, but in that case, signed type shape should rebind to unsigned type with same size.

I think it's safe and simpler to always use `rebind_container_t`

Reply:
@JohanMabille  If I call it directly I will get a compiler error on `xt::xfixed_index<size_t...>` thus I'm trying to SFINAE those calls out.

Reply:
Oooh, I had totally forgot this one. But I think it is not very explicitly disabled here (and I hav the feeling this works only for `xfixed` with unsigned values). A more explicit way would be to specialize the `rebind_shape` structure for `xfixed`. The generic implementation can "forward" the call to rebind_container_t.

Reply:
@JohanMabille  I can do that. It's more obvious for sure.

Reply:
done!